tests/envs/test_satellites.py

```diff
@@ -0,0 +1,155 @@
+"""Test cases for the satellites environment."""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+from predicators.src import utils
+from predicators.src.envs.satellites import SatellitesEnv
+from predicators.src.structs import Action
+
+
+def test_satellites():
+    """Tests for SatellitesEnv()."""
+    utils.reset_config({
+        "env": "satellites",
+        "satellites_num_sat_train": [1],
+        "satellites_num_obj_train": [1],
+        "satellites_num_sat_test": [3],
+        "satellites_num_obj_test": [3],
+    })
+    env = SatellitesEnv()
+    for task in env.get_train_tasks():
+        for obj in task.init:
+            assert len(obj.type.feature_names) == len(task.init[obj])
+    for task in env.get_test_tasks():
+        for obj in task.init:
+            assert len(obj.type.feature_names) == len(task.init[obj])
+    assert len(env.predicates) == 13
+    assert len(env.goal_predicates) == 3
+    assert len(env.options) == 5
+    assert len(env.types) == 2
+    obj_type, sat_type = sorted(env.types)
+    assert obj_type.name == "object"
+    assert sat_type.name == "satellite"
+    assert env.action_space.shape == (10, )
+    task = env.get_test_tasks()[0]
+    state = task.init
+    sat0, sat1, sat2 = state.get_objects(sat_type)
+    obj0, obj1, obj2 = state.get_objects(obj_type)
+    assert sat0.name == "sat0"
+    assert sat1.name == "sat1"
+    assert sat2.name == "sat2"
+    assert obj0.name == "obj0"
+    assert obj1.name == "obj1"
+    assert obj2.name == "obj2"
+    state.set(sat1, "is_calibrated", 1.0)
+    state.set(sat2, "read_obj_id", 0.0)
+    state.set(obj0, "has_chem_x", 1.0)
+    state.set(obj1, "has_chem_y", 1.0)
+    state.set(obj2, "has_chem_x", 1.0)
+    state.set(obj2, "has_chem_y", 1.0)
+    # Some arcane errors arise with calling fig2data() during unit testing,
```

Comment:
curious what exactly this is

Reply:
```
        # Some arcane errors arise with calling fig2data() during unit testing,
        # so we'll just call render_state_plt() instead of render_state() here.
>       env.render_state(state, task)

tests/envs/test_satellites.py:53:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/envs/base_env.py:129: in render_state
    img = utils.fig2data(fig, dpi=CFG.render_state_dpi)
src/utils.py:2179: in fig2data
    fig.canvas.draw()
../../.virtualenvs/igibson/lib/python3.9/site-packages/matplotlib/backends/backend_agg.py:406: in draw
    self.figure.draw(self.renderer)
../../.virtualenvs/igibson/lib/python3.9/site-packages/matplotlib/artist.py:74: in draw_wrapper
    result = draw(artist, renderer, *args, **kwargs)
../../.virtualenvs/igibson/lib/python3.9/site-packages/matplotlib/artist.py:51: in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
../../.virtualenvs/igibson/lib/python3.9/site-packages/matplotlib/figure.py:2790: in draw
    mimage._draw_list_compositing_images(
../../.virtualenvs/igibson/lib/python3.9/site-packages/matplotlib/image.py:132: in _draw_list_compositing_images
    a.draw(renderer)
../../.virtualenvs/igibson/lib/python3.9/site-packages/matplotlib/artist.py:51: in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
../../.virtualenvs/igibson/lib/python3.9/site-packages/matplotlib/_api/deprecation.py:431: in wrapper
    return func(*inner_args, **inner_kwargs)
../../.virtualenvs/igibson/lib/python3.9/site-packages/matplotlib/axes/_base.py:2921: in draw
    mimage._draw_list_compositing_images(renderer, self, artists)
../../.virtualenvs/igibson/lib/python3.9/site-packages/matplotlib/image.py:132: in _draw_list_compositing_images
    a.draw(renderer)
../../.virtualenvs/igibson/lib/python3.9/site-packages/matplotlib/artist.py:51: in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)
../../.virtualenvs/igibson/lib/python3.9/site-packages/matplotlib/text.py:679: in draw
    bbox, info, descent = textobj._get_layout(renderer)
../../.virtualenvs/igibson/lib/python3.9/site-packages/matplotlib/text.py:306: in _get_layout
    _, lp_h, lp_d = renderer.get_text_width_height_descent(
../../.virtualenvs/igibson/lib/python3.9/site-packages/matplotlib/backends/backend_agg.py:239: in get_text_width_height_descent
    font = self._get_agg_font(prop)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <matplotlib.backends.backend_agg.RendererAgg object at 0x136f99610>, prop = <matplotlib.font_manager.FontProperties object at 0x136f99b50>

    def _get_agg_font(self, prop):
        """
        Get the font for text instance t, caching for efficiency
        """
        fname = findfont(prop)
        font = get_font(fname)

        font.clear()
        size = prop.get_size_in_points()
>       font.set_size(size, self.dpi)
E       RuntimeError: In set_size: Could not set the fontsize (error code 0x97)

../../.virtualenvs/igibson/lib/python3.9/site-packages/matplotlib/backends/backend_agg.py:278: RuntimeError
```

Reply:
it has to do with the combination of (1) the backend is different during unit testing for some reason, (2) this env is the first time we are using `ax.text` in the plt, which is what's crashing here